File: 5course/BigData/lab2/mapper.js
#!/usr/bin/env node

const readline = require("readline")

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
})

const stopwords = ["a", "and", "around", "every", "for", "from", "in",
"is", "it", "not", "on", "one", "the", "to", "under", "of", "at", "are", "as", "or"]

rl.on("line", line => {
    if (!line) return

    line.split(" ").map(word => {
        const cleanWord = word.replace(/[^a-zA-Z]/g, "")
        if (stopwords.includes(cleanWord)) return

        console.log(`${cleanWord}\t1`)
    })
})

rl.on("close", () => {
    process.exit(0)
})

File: 5course/BigData/lab2/mapper.py
#!/usr/bin/env python

stopwords = ["a", "and", "around", "every", "for", "from", "in",
"is", "it", "not", "on", "one", "the", "to", "under", "of", "at", "are", "as", "or"]

import sys
import re

for line in sys.stdin:
    words = line.split()
    for word in words:
        cleanWord = re.sub('\W+', '', word)
        if (cleanWord in stopwords):
            continue
        print('%s\t%s' % (cleanWord, 1))